zapcli/zap_helper.py

```diff
@@ -343,6 +348,170 @@ def html_report(self, file_path):
+            if key in config_dict and config_dict[key] == 'FAIL':
+                user_msg = ''
+                if key in config_msg:
+                    user_msg = config_msg[key]
+                self.__print_rule(config_dict[key], alert_list, detailed_output, user_msg, min_level)
+                fail_count += 1
+
+        self.logger.info('FAIL: ' + str(fail_count) + '\tWARN: ' + str(warn_count) + '\tINFO: ' + str(info_count) +
+                         '\tIGNORE: ' + str(ignore_count) + '\tPASS: ' + str(pass_count))
+
+        return (fail_count, warn_count, info_count, ignore_count, pass_count)
+
+    def __load_config(self, config):
+        """Load and parse configuration"""
+        config_dict = {}
+        config_msg = {}
+        for line in config:
+            if not line.startswith('#') and len(line) > 1:
+                (key, val, optional) = line.rstrip().split('\t', 2)
+                if key == 'OUTOFSCOPE':
+                    for plugin_id in val.split(','):
+                        if plugin_id not in self.out_of_scope_dict:
+                            self.out_of_scope_dict[plugin_id] = []
+                            self.out_of_scope_dict[plugin_id].append(re.compile(optional))
+                else:
+                    config_dict[key] = val
+                    if '\t' in optional:
+                        (_, usermsg) = optional.rstrip().split('\t')
+                        config_msg[key] = usermsg
+                    else:
+                        config_msg[key] = ''
+
+        return (config_dict, config_msg)
+
+    def __get_all_alerts(self):
+        """Return all the alerts (using paging) from zap"""
+        st = 0
+        pg = 100
+        alert_dict = {}
+        alerts = self.zap.core.alerts(start=st, count=pg)
+
+        while len(alerts) > 0:
+            for alert in alerts:
+                plugin_id = alert.get('pluginId')
+                if plugin_id in self.blacklist:
+                    continue
+                if not self.__is_in_scope(plugin_id, alert.get('url')):
+                    continue
+                if plugin_id not in alert_dict:
+                    alert_dict[plugin_id] = []
+                alert_dict[plugin_id].append(alert)
+            st += pg
+            alerts = self.zap.core.alerts(start=st, count=pg)
+
+        return alert_dict
+
+    def __is_in_scope(self, plugin_id, url):
+        """Check if the given url and plugin is in scope"""
+        if '*' in self.out_of_scope_dict:
+            for oos_prog in self.out_of_scope_dict['*']:
+                if oos_prog.match(url):
+                    return False
+        if plugin_id in self.out_of_scope_dict:
+            for oos_prog in self.out_of_scope_dict[plugin_id]:
+                if oos_prog.match(url):
+                    return False
+        return True
+
+    def __print_rule(self, action, alert_list, detailed_output, user_msg, min_level):
```

Comment:
Ok, I took another look and it seems to me that it will be hard to achieve that. What do you think about extracting the whole logic to the cli? The helper will only contain the code responsible for exporting the alert - which already exist.